shared/chat/conversation/list/index.desktop.js

```diff
@@ -1,244 +1,269 @@
-      }
-    })
+  _onRowsRendered = ({stopIndex}: {stopIndex: number}) => {
+    this._lastRowIdx = stopIndex
+  }
+
+  _setListRef = (r: any) => {
+    this._list = r
   }
 
-  _updateInternalMessages = (props: Props) => {
-    if (props.messages !== this.state.messages) {
-      this._invalidateChangedMessages(props)
-      this.setState({
-        messages: props.messages,
-      })
+  render () {
+    if (!this.props.validated) {
+      return (
+        <div style={{alignItems: 'center', display: 'flex', flex: 1, justifyContent: 'center'}}>
+          <Icon type='icon-securing-266' style={{alignSelf: 'flex-start'}} />
+        </div>
+      )
     }
+
+    const rowCount = this.props.messageKeys.count()
+    const scrollToIndex = this.state.isLockedToBottom ? rowCount - 1 : this._keepIdxVisible
+
+    // We pass additional props (listRerender, selectedMessageKey) to Virtualized.List so we can force re-rendering automatically
+    return (
+      <div style={containerStyle} onClick={this._handleListClick} onCopyCapture={this._onCopyCapture}>
+        <style>{realCSS}</style>
+        <Virtualized.AutoSizer onResize={this._onResize}>
+          {({height, width}) => (
+            <Virtualized.List
+              messageKeys={this.props.messageKeys}
+              listRerender={this.state.listRerender}
+              selectedMessageKey={this.state.selectedMessageKey}
+              columnWidth={width}
+              deferredMeasurementCache={this._cellCache}
+              height={height}
+              onScroll={this._onScroll}
+              onRowsRendered={this._onRowsRendered}
+              ref={this._setListRef}
+              rowCount={rowCount}
+              rowHeight={this._cellCache.rowHeight}
+              rowRenderer={this._rowRenderer}
+              scrollToAlignment='end'
+              scrollToIndex={scrollToIndex}
+              style={listStyle}
+              width={width}
+            />
+          )}
+        </Virtualized.AutoSizer>
+      </div>
+    )
   }
+}
 
-  _onScrollSettled = _.debounce(() => {
-    // If we've stopped scrolling let's update our internal messages
-    this._updateInternalMessages(this.props)
-    this.setState({
-      isScrolling: false,
-    })
-  }, 1000)
+// We need to use both visibility and opacity css properties for the
+// action button hide/show on hover.
+// We use opacity because it shows/hides the button immediately on
+// hover, while visibility has slight lag.
+// We use visibility so that the action button content isn't copied
+// during copy/paste actions since user-select isn't working in
+// Chrome.
+const realCSS = `
+.message {
+  border: 1px solid transparent;
+}
+.message .action-button {
+  visibility: hidden;
+  opacity: 0;
+}
+.message:hover {
+  border: 1px solid ${globalColors.black_10};
+}
+.message:hover .action-button {
+  visibility: visible;
+  opacity: 1;
+}
+`
+
+// Adds in popup handling
+class PopupEnabledList extends BaseList {
```

Comment:
I moved the popup handling stuff here as its all sorta related and unrelated to the core list handling stuff above